wcfsetup/install/files/lib/data/I18nDatabaseObjectList.class.php

```diff
@@ -0,0 +1,54 @@
+<?php
+namespace wcf\data;
+use wcf\system\WCF;
+use wcf\util\StringUtil;
+
+/**
+ * Abstract class for a list of database objects with better sorting of i18n-based columns.
+ *
+ * @author	Florian Gail
+ * @copyright	2001-2017 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\Data
+ * @since	3.1
+ */
+abstract class I18nDatabaseObjectList extends DatabaseObjectList {
+	/**
+	 * Array of fields filled with language items
+	 * this slows down the system, use this only when sorting
+	 * based on an i18n-value.
+	 * The value represents the original field name.
+	 * The key represents the new field name and is optional.
+	 * If no key is provided, the new field will be 'i18n'.$value
+	 * 
+	 * use table.field schema!
+	 * 
+	 * @example   [ 'title' => 'titleSortField' ]
+	 * 
+	 * @var string[]
+	 */
+	public $i18nFields = [];
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function readObjects() {
+		if (!empty($this->i18nFields)) {
+			foreach ($this->i18nFields as $key => $value) {
+				$matchTable =  'i18n_' . StringUtil::getHash($value);
+				
+				if (!empty($this->sqlSelects)) $this->sqlSelects .= ", ";
+				
+				if (is_numeric($key)) {
```

Comment:
I'm not sure about this. I feel that an explicit name should be given in all cases.